pkg/util/steps/condition.go

```diff
@@ -66,9 +86,29 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
```

Comment:
This utility function `c.enrichConditionTimeoutError()` might be easier to test if placed outside of `c`. You could set up something like the below, which passes `conditionFunction` as a parameter to utility function `enrichConditionTimeoutError()` that then provides the expected error. 
```suggestion
		return enrichConditionTimeoutError(c.f)
```
Functionally it would be the same but the unit test would be simpler.

Reply:
Done.